src/components/SuperDrawer/useSuperDrawer.tsx

```diff
@@ -110,6 +118,10 @@ export function useSuperDrawer(): UseSuperDrawerHook {
           // Reset Modal state
           modalState.current = undefined;
         }
+        const maybeValidated = validateDrawerChange(onClose);
+        if (maybeValidated !== undefined) {
+          return maybeValidated
+        }
 
         onClose();
```

Comment:
I.e. I think this "extra" onClose call could go away because line 121 would be just `validateDrawerChange(onClose)`